app/components/catalog-settings/component.js

```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+
+  actions: {
+    keymapChanged: function() {
+      this.sendAction('keymapChanged', this.get('keymap.asMap'));
+    },
+  },
+
+  keymap: Ember.Object.create({}),
+
+  didInitAttrs: function() {
+
+    var catalogs = this.get('catalog.value');
+    var catalogsSplit = catalogs.split(',');
+    var keymapOut = {};
+
+    if (catalogsSplit.length > 1) {
+
+      // its not default so we need to parse it
+      catalogsSplit.forEach((item) => {
+        var split = item.split('=');
+        keymapOut[split[0]] = split[1];
+      });
+
+      this.set('keymap', keymapOut);
+    } else {
+
+      var catalogItem = catalogsSplit[0];
+
+      if (catalogItem.indexOf('=')) {
+
+        var split = catalogItem.split('=');
+        var obj = {};
+
+        obj[split[0]] = split[1];
+
+        this.set('keymap', obj);
+      } else {
+
+        this.set('keymap', {'library': catalogsSplit[0]});
+      }
+    }
+  },
+
+  keymapObserver: function() {
+    this.send('keymapChanged');
```

Comment:
Since the `keymapChanged` action just calls `sendAction` you could remove it and just call `sendAction` here..